reframe/utility/typecheck.py

```diff
@@ -424,3 +411,50 @@ def make_meta_type(name, cls, metacls=_BuiltinType):
 Set     = make_meta_type('Set', set, _SequenceType)
 Str     = make_meta_type('Str', str, _StrType)
 Tuple   = make_meta_type('Tuple', tuple, _TupleType)
+
+
+class Duration(metaclass=ConvertibleType):
+    '''A float type that represents duration in seconds.
+
+    This type supports the following implicit conversions:
+
+    - From integer values
+    - From string values in the form of ``<days>d<hours>h<minutes>m<seconds>s``
+
+    .. versionadded:: 4.2
+
+    '''
+
+    def _assert_non_negative(val):
+        if val < 0:
+            raise ValueError('duration cannot be negative')
+
+        return val
+
+    @classmethod
+    def __rfm_cast_int__(cls, val):
+        return Duration._assert_non_negative(float(val))
+
+    @classmethod
+    def __rfm_cast_float__(cls, val):
+        return Duration._assert_non_negative(val)
+
+    @classmethod
+    def __rfm_cast_str__(cls, val):
+        # First try to convert to a float
+        try:
+            val = float(val)
+        except ValueError:
+            time_match = re.match(r'^((?P<days>\d+)d)?'
```

Comment:
@vkarak Do you think we should allow floating point durations as well? If so the regex `\d*\.*\d+` should be used. I am asking since timedelta accepts floating point values.

Reply:
You can still set it as a float, either from the command line or from within the test. This checks only the case when you assign a string to it. So there we don't allow `3.5d`. I haven't actually changed that from the `TimerField`.